src/suggestion-box.ts

```diff
@@ -0,0 +1,383 @@
+import { LitElement, html, css } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import { queryAssignedElements } from "lit/decorators.js";
+
+@customElement("aalam-sgn-box")
+export class SuggestionBox extends LitElement {
+    static styles = css`
+        :host {
+            display: block;
+            position: relative;
+        }
+    `;
+
+    @property({ type: Array<Object> })
+    list: Array<{ [key: string]: any }> = [];
+
+    @property()
+    listkey: string = "";
+
+    @property()
+    slot_input = null;
+
+    @property()
+    _private_slot_el: boolean = false;
+
+    @queryAssignedElements({ slot: "sgn-item-template" })
+    template_slots: Array<HTMLElement>;
+
+    @queryAssignedElements({ slot: "__private-item" })
+    private_items: Array<HTMLElement>;
+
+    @property()
+    prevItem: number = 0;
+
+    @state()
+    has_more: boolean = false;
+
+    @state()
+    index: number = -1;
+
+    @property()
+    size: number = 0;
+
+    @state()
+    show_empty: boolean = true;
+
+    @property()
+    result: Array<{ [key: string]: string }> = [];
+
+    @property()
+    highlight: string = "";
+
+    @property()
+    curr_index: number = 0;
+
+    @property()
+    input_el: HTMLElement | null = null;
+
+    @property()
+    filtered_list: Array<{ [key: string]: string }> = [];
+
+    @property()
+    prevIndex: number = 0;
+
+    @state()
+    show_container = false;
+
+    private _outClickListener = this.windowClickEvent.bind(this);
+    private input_temp: HTMLElement | null | string = null;
```

Comment:
Used in only one function. This can be declared as a local variable in that function